Fix handling of buckets with disabled bucket notifications

welansari opened the pull request:
Backport https://github.com/scality/backbeat/pull/2655

Issue: BB-697

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-697 contains:


* `9.0.13`

* `9.1.0`



Considering where you are trying to merge, I expected to find at least:



* `9.0.4.2`


Please check the `Fix Version/s` of BB-697, or the target
branch of this pull request.

williamlardier reviewed (approved):
Please backport the changes before merging :slightly_smiling_face:

Commit:
Fix error handling fallback in _processObjectEntry

Added try/catch to _processObjectEntry to catch exception
and log them before returning an error

Issue: BB-697

Commit:
Fix handling of disabled bucket notifications

When bucket notifications are disabled after being enabled,
the configuration is set to an empty object. Previously, this
case was not handled properly, leading to crashes as the
notification producer and processor incorrectly considered the
configuration available. This fix ensures proper handling of
empty configurations to prevent such issues.

Issue: BB-697

welansari commented:
~~Closing PR as no release of ARTESCA 3.1.3 is currently planned.~~

Commit:
bump version to 9.0.4.2

Issue: BB-697

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

welansari commented:
/approve

bert-e commented:
# Build failed



The [build](https://github.com/scality/backbeat/actions?query=branch%3Abugfix/BB-697-backport-9.0.4) for [commit](https://github.com/scality/backbeat/commit/096dbd3c9acb05cb71255bce549780582c5ed350) did not succeed in branch [bugfix/BB-697-backport-9.0.4](https://github.com/scality/backbeat/compare/hotfix/9.0.4...bugfix/BB-697-backport-9.0.4)



*The following options are set:* **approve**

Commit:
fix ft tests

Debian Buster repositories are no longer available.
Debian Buster reached end-of-life and has been moved to archive repositories

Issue: BB-697

bert-e commented:
# Build failed



The [build](https://github.com/scality/backbeat/actions?query=branch%3Abugfix/BB-697-backport-9.0.4) for [commit](https://github.com/scality/backbeat/commit/21cb21f8d3afda25ac80e07a223df6d729737290) did not succeed in branch [bugfix/BB-697-backport-9.0.4](https://github.com/scality/backbeat/compare/hotfix/9.0.4...bugfix/BB-697-backport-9.0.4)



*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `hotfix/9.0.4`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.6`
* `development/9.0`
* `development/9.1`



Please check the status of the associated issue BB-697.

Goodbye kerkesni.




*The following options are set:* **approve**
